io/plugins/eu.esdihumboldt.hale.io.haleconnect.ui/src/eu/esdihumboldt/hale/io/haleconnect/ui/projects/OwnerFilterEntry.groovy

```diff
@@ -16,14 +16,17 @@
 package eu.esdihumboldt.hale.io.haleconnect.ui.projects
 
 import eu.esdihumboldt.hale.io.haleconnect.Owner
-import groovy.transform.Immutable
-
+import groovy.transform.Canonical
 /**
  * Entry in the owner filter dropdown of LoadHaleConnectProjectWizardPage
+ *
+ * After migration to java17, if this class is marked as @Immutable then 
+ * at runtime an exception is thrown indicating that clone() is not applicable to Owner class.
+ * Therefore, using @Canonical which provides similar functionality but immutability. 
  * 
  * @author Florian Esser
  */
-@Immutable
+@Canonical
```

Comment:
Can we compromise on immutability here?
https://docs.groovy-lang.org/2.4.7/html/gapi/groovy/transform/Immutable.html

Reply:
Based on the documentation I would guess that since `clone` is used for arrays this causes a problem, since `Owner` does not implement it.

I would see two options:

- let `Owner` implement `Cloneable` (e.g. via [@AutoClone](https://docs.groovy-lang.org/next/html/gapi/groovy/transform/AutoClone.html))
- use a collection type (e.g. `List`) instead of an array for `owner`

I would prefer using `@AutoClone` if that works.